src/store/priceList/actions.ts

```diff
@@ -14,12 +16,20 @@ export const {
   'priceList/fetch/failure'
 )<void, Rates, AxiosError>();
 
-export function fetchPriceList(providerUuid) {
+export function fetchPriceList(providerUuid, time = 0) {
+  const TTL = time * 1000 * 60; // convert from minutes to milliseconds
   return (dispatch: Dispatch, getState) => {
     dispatch(fetchPriceListRequest());
+    const rates = cache[providerUuid];
+    const now = new Date();
+    if (rates && now.valueOf() - rates.time.valueOf() < TTL) {
+      dispatch(fetchPriceListSuccess(rates.data));
+      return;
+    }
     return fetchRate(providerUuid)
       .then(res => {
         dispatch(fetchPriceListSuccess(res.data));
+        cache[providerUuid] = { data: res.data, time: new Date() };
```

Comment:
Please take a look at how we fetch report in `awsReportsActions`, `ocpReportsActions`, etc. Although I didn't create the code initially, I believe that's a best practice approach for caching data.

In that code, we don't cache in awsReportsActions, we simply skip making a request for another report. Ideally, the reducer would cache the report, allowing the selector to return any stored report based on the `providerUuid` -- similar to how we provide unique IDs based on the widget ID / query.

Note that the `awsReportsActions` code below fetches a new report every 30 minutes.

```
export function fetchReport(
  reportType: AwsReportType,
  query: string
): ThunkAction<void, RootState, void, any> {
  return (dispatch, getState) => {
    if (!isReportExpired(getState(), reportType, query)) {
      return;
    }
...
```
    }

```
function isReportExpired(
  state: RootState,
  reportType: AwsReportType,
  query: string
) {
  const report = selectReport(state, reportType, query);
  const fetchStatus = selectReportFetchStatus(state, reportType, query);
  if (fetchStatus === FetchStatus.inProgress) {
    return false;
  }

  if (!report) {
    return true;
  }

  const now = Date.now();
  return now > report.timeRequested + expirationMS;
}
```

While the `awsReportsSelectors` code is able to obtain any cached report based on a unique ID / query.

```
export const selectReport = (
  state: RootState,
  reportType: AwsReportType,
  query: string
) => selectReportsState(state).byId.get(getReportId(reportType, query));
```

Finally, the `awsReortsReducer` code caches the report so the selector can retrieve by ID.

```
case getType(fetchAwsReportSuccess):
      return {
        ...state,
        fetchStatus: new Map(state.fetchStatus).set(
          action.meta.reportId,
          FetchStatus.complete
        ),
        byId: new Map(state.byId).set(action.meta.reportId, {
          ...action.payload,
          timeRequested: Date.now(),
        }),
        errors: new Map(state.errors).set(action.meta.reportId, null),
      };
```